catkin_ws/src/state_estimation/src/sensors.py

```diff
@@ -19,111 +18,121 @@
-        if rospy.get_time() == 0: return False
-        if not self.hasValidData(): return False #check that state is complete
-        if rospy.get_time() - self.last_unique_state_time > self.time_before_considered_inactive: #check that state has changed in last N seconds
-            if rospy.get_time() - self.last_error_message_time > 1:
-                self.last_error_message_time = rospy.get_time()
-                rospy.logwarn("{} has been inactive for {} seconds.".format(self.sensor_name, self.time_before_considered_inactive))
-            return False
-        else:
-            return True
+    def get_is_active(self):
+        return self.is_active
 
-    def hasValidData(self):
-        raise NotImplementedError    
 
+# Depth Sensor class inheriting from Sensor
 class DepthSensor(Sensor):
     def __init__(self):
         super().__init__("Depth Sensor")
+        
+        self.z = 0
         self.z_pos_mount_offset = 0
+
+        rospy.Subscriber("/sensors/depth/status", Int32, self.sensor_status_cb)
         rospy.Subscriber("/sensors/depth/z", Float64, self.depth_cb)
 
-    def depth_cb(self, depth_msg):
-        self.z = depth_msg.data + self.z_pos_mount_offset
-        self.updateLastState()
-    
-    def hasValidData(self):
-        return self.z is not None
+    def depth_cb(self, msg):
+        self.z = msg.data + self.z_pos_mount_offset
+
+    def get_reading(self):
+        return {
+            "z": self.z
+        }
 
+
+# IMU class inheriting from Sensor
 class IMU(Sensor):
     def __init__(self):
         super().__init__("IMU")
-
+        
         q_imunominal_imu_w = rospy.get_param("q_imunominal_imu_w")
         q_imunominal_imu_x = rospy.get_param("q_imunominal_imu_x")
         q_imunominal_imu_y = rospy.get_param("q_imunominal_imu_y")
         q_imunominal_imu_z = rospy.get_param("q_imunominal_imu_z")
 
         self.q_imunominal_imu = np.quaternion(q_imunominal_imu_w, q_imunominal_imu_x, q_imunominal_imu_y, q_imunominal_imu_z)
         self.q_imu_auv = self.q_imunominal_imu.conjugate() * Q_IMUNOMINAL_AUV
-        self.q_nwu_auv = None
-        
+        self.q_nwu_auv = np.quaternion(1, 0, 0, 0)
+
+        self.angular_velocity = np.array([0,0,0])
+
+        rospy.Subscriber("/sensors/imu/status", Int32, self.sensor_status_cb)
         rospy.Subscriber("/sensors/imu/angular_velocity", SbgImuData, self.ang_vel_cb)
         rospy.Subscriber("/sensors/imu/quaternion", SbgEkfQuat, self.quat_cb)
-        
+
     def ang_vel_cb(self, msg):
         # angular velocity vector relative to imu frame 
         ang_vel_imu = np.array([msg.gyro.x, msg.gyro.y, msg.gyro.z])
         # anuglar velocity vector relative to AUV frame 
         temp_angular_velocity = self.q_imu_auv * np.quaternion(0, ang_vel_imu[0], ang_vel_imu[1], ang_vel_imu[2]) * self.q_imu_auv.conjugate()
         self.angular_velocity = np.array([temp_angular_velocity.x, temp_angular_velocity.y, temp_angular_velocity.z])
-        self.updateLastState()
```

Comment:
no need to update last reading given that the callback functions already do that